### PR TITLE
support `LargeList` in `array_prepend` and `array_append`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Parts #8185 

## What changes are included in this PR?

- Support `LargeList` in `array_prepend` and `array_append`
- Support cast LargeList in `ArrayAndElement`

### CODE REVIEW

File: datafusion/physical-expr/src/array_expressions.rs [hunk 1]
```diff
@@ -52,22 +52,6 @@ macro_rules! downcast_arg {
...
-    ($ARGS:expr, $ARRAY_TYPE:ident) => {{
-        $ARGS
-            .iter()
-            .map(|e| match e.as_any().downcast_ref::<$ARRAY_TYPE>() {
-                Some(array) => Ok(array),
-                _ => internal_err!("failed to downcast"),
-            })
-    }};
-}
-
```
Comment (Weijun-H) [inline datafusion/physical-expr/src/array_expressions.rs:70]: Unused function

File: datafusion/sqllogictest/test_files/array.slt [hunk 2]
```diff
@@ -107,6 +107,18 @@ AS VALUES
...
+# AS
+#   SELECT
+#     arrow_cast(column1, 'LargeList(LargeList(Int64))') AS column1,
+#     arrow_cast(column2, 'LargeList(Int64)') AS column2,
+#     column3,
+#     arrow_cast(column4, 'LargeList(LargeList(List(Int64)))') AS column4,
+#     arrow_cast(column5, 'LargeList(Int64)') AS column5
+#   FROM nested_arrays
+# ;
+
```
Comment (Weijun-H) [inline datafusion/sqllogictest/test_files/array.slt:121]: cast `List(List)` to `LargeList(LargeList)` are not supported yet.

Comment (comphead) [reply datafusion/sqllogictest/test_files/array.slt:121]: Please add the comment to the test why it is commented out

File: datafusion/sqllogictest/test_files/array.slt [hunk 3]
```diff
@@ -131,6 +154,16 @@ AS VALUES
   (NULL, NULL, NULL, NULL)
 ;
 
+# statement ok
+# CREATE TABLE large_arrays_values_v2
+# AS SELECT
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/array.slt:159]: same

**Review by comphead — COMMENTED** [review]
looks good @Weijun-H please add comments to the disabled tests, why exactly it got disabled

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @Weijun-H

**Pull request merged**